public/app/core/components/NavBar/NavBarNext.tsx

```diff
@@ -48,6 +40,7 @@ export const NavBarNextUnconnected = React.memo(({ navBarTree }: Props) => {
   };
   const navTree = cloneDeep(navBarTree);
   const coreItems = navTree.filter((item) => item.section === NavSection.Core);
+  const pinnedItems = navTree.filter((item) => item.showInNavBar);
```

Comment:
instead of creating a `pinnedItems` array, i think we could change the filters for `coreItems`/`pluginItems`/`configItems` to include this filter. something like:
```javascript
const coreItems = navTree.filter((item) => (item.section === NavSection.Core) && item.showInNavBar);
```